src/dev/src/app/datalist/datalist.demo.html

```diff
@@ -0,0 +1,35 @@
+<!--
+  ~ Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+<h2>Vertical</h2>
+<form clrForm clrLayout="vertical" (submit)="submit()" [formGroup]="model">
+  <clr-datalist-container>
+    <label>Datalist - Reactive Form</label>
+    <input clrDatalist [clrData]="data" placeholder="Option" name="Option" autocomplete="off" formControlName="item" />
+  </clr-datalist-container>
+  <button class="btn btn-primary" type="submit" [disabled]="model.invalid">Submit</button>
+</form>
+
+<h2>Horizontal <button class="btn btn-sm btn-primary" (click)="disabled = !disabled">Toggle Disabled</button></h2>
+<form clrForm clrLayout="horizontal">
+  <clr-datalist-container>
+    <input clrDatalist [clrData]="data" [(ngModel)]="vertical.one" placeholder="No label, wrapper" name="Option2"/>
+  </clr-datalist-container>
+
+  <clr-datalist-container>
+    <label>Full example</label>
+    <input clrDatalist [clrData]="data" [(ngModel)]="vertical.two" placeholder="Full example" required name="Option3" minlength="8" />
```

Comment:
Instead of passing in a `Input` of `clrData` we can keep the datalist element ouside of the component and allow the consumer to declarativley create the list. We typically will try to keep things declarative if possible since it gives the consumer more flexibility.


```html
<clr-datalist-container>
  <label>Full example</label>
  <input clrDatalist [(ngModel)]="vertical.two" placeholder="Full example" required name="Option3" minlength="8"  />
  <datalist>
    <option *ngFor="let item of items" [value]="item"></option>
  </datalist>
</clr-datalist-container>
```

Reply:
This can be done. My first thought was that having the datalist into container component would be easier to use on the consumer's side and i would have more control over how the datalist and input are linked through a preset id. My question now is whether we will have to let the consumer set the ids himself or how should we proceed?

Reply:
You can follow a similar pattern to the other inputs which auto preset the id if a custom id is not present.